### PR TITLE
refactor map to use child components

### PR DESCRIPTION
**Pull request by maityarkodeb**
Refactoring of components to use child context.

### CODE REVIEW

File: packages/react-ui-map/src/Mapbox/MapLayer.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export default class MapLayer extends Component {
+  static propTypes = {
+    layers: PropTypes.array
+  }
+
+  static contextTypes = {
+    map: PropTypes.object
+  }
+
+  componentWillMount() {
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/MapLayer.js]: Can you place the code from `componentWillMount` ([from docs](https://facebook.github.io/react/docs/react-component.html#componentwillmount)) in the `constructor` instead?

File: packages/react-ui-map/src/Mapbox/MapLayer.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    const { map } = this.context
+    const { layers } = this.props
+
+    for(let value of layers) {
+        map.addLayer(value.layer)
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/MapLayer.js]: Please also ensure `map` is available in order to call a method on it.

File: packages/react-ui-map/src/Mapbox/MapSource.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  static propTypes = {
+    sources: PropTypes.array,
+    children: PropTypes.node
+  }
+
+  static contextTypes = {
+    map: PropTypes.object
+  }
+
+  componentWillMount() {
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/MapSource.js]: Same as the comment above about `componentWillMount`.

File: packages/react-ui-map/src/Mapbox/MapSource.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+export default class MapSource extends Component {
+  static propTypes = {
+    sources: PropTypes.array,
+    children: PropTypes.node
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/MapSource.js]: Where is `children` being used here?

File: packages/react-ui-map/src/Mapbox/index.js [hunk 5]
```diff
@@ -51,14 +47,7 @@ export default class Mapbox extends PureComponent {
...
     map.on('load', () => {
-      if (this.state.map !== map) {
-        this.setState(map)
-      }
-
-      sources.forEach((v, i) => {
-        map.addSource(sources[i].id, sources[i].source)
-        map.addLayer(layers[i].layer)
-      })
+      this.setState({ map })
```
Comment (tadjohnston) [inline packages/react-ui-map/src/Mapbox/index.js]: Does this need to be set on load here or can it be set directly after the instantiation for `new mapboxgl.Map()`?

File: packages/react-ui-map/src/index.js [hunk 6]
```diff
@@ -1,2 +1,4 @@
 export { default } from './Mapbox'
 export { default as Mapbox } from './Mapbox'
+export { default as MapSource } from './Mapbox/MapSource'
```
Comment (tadjohnston) [inline packages/react-ui-map/src/index.js]: I think we should probably change the names to `Source` and `Layer` for the filenames.  This is kinda nitpicky so it's up to what others ideas are.

File: packages/react-ui-map/src/Mapbox/MapSource.js (hunk 4 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/MapSource.js]: `children` was not being used.  removed.

File: packages/react-ui-map/src/Mapbox/MapLayer.js (hunk 1 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/MapLayer.js]: Done.

File: packages/react-ui-map/src/index.js (hunk 6 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/index.js]: Changed names to `Source` and `Layer`.

File: packages/react-ui-map/src/Mapbox/index.js (hunk 5 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/index.js]: Leaving in on load as per discussion with @tadjohnston.

File: packages/react-ui-map/src/Mapbox/MapSource.js (hunk 3 above)
Comment (maityarkodeb) [inline packages/react-ui-map/src/Mapbox/MapSource.js]: Addressed.

File: packages/react-ui-map/src/Mapbox/Layer.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+export default class Layer extends Component {
+  constructor(props, context) {
+    super(props, context)
+
+    const { map } = this.context
+    const { layers } = this.props
+
+    for(let value of layers) {
+        map.addLayer(value.layer)
+    }
```
Comment (lfender6445) [inline packages/react-ui-map/src/Mapbox/Layer.js]: fix indentation for files please, use linter

**Comment by lfender6445** [comment]
👍

**Pull request merged**